spectral_cube/tests/test_masks.py

```diff
@@ -543,3 +542,16 @@ def test_filled():
     assert_allclose(filled, filled_)
 
     assert (np.isnan(filled) == mcube.mask.exclude()).all()
+
+def test_boolean_array_composite_mask():
+
+    cube, data = cube_and_raw('adv.fits')
+
+    med = cube.median()
+    mask = cube > med
+    arrmask = cube.max(axis=0) > med
+
+    # we're just testing that this doesn't fail
+    combined_mask = mask & arrmask
+
+    mcube = cube.with_mask(combined_mask)
```

Comment:
Add some sort of operation here that will trigger a call to `mask.include` to make sure the mask has the correct WCS object. I don't think `validate_wcs` will be called in the mask until an operation is performed on the masked cube.